### ISSUE TITLE
Missing adversarial loss

### ISSUE THREAD

**Issue by fa9r** [aayushbansal/Recycle-GAN#12]
Hey, I think I found a bug in your adversarial loss of both models (RecycleGAN and ReCycleGAN). The adversarial loss for D is calculated on G_y(x_t-i), P_y(G_y(x_t-i)) and P_y(y_t-i), but the loss for G/P is only calculated on G_y(x_t-i) and P_y(G_y(x_t-i)). I.e. P does not learn from P_y(y_t-i), since the GAN loss for pred_A2 and pred_B2 is missing in backward_G().

If this is in fact a bug, I can quickly fix it and make a pull request if you want. I already fixed it in my own implementation, but I have not compared it to the original version yet.

**Comment by aayushbansal** [comment]
You are correct.  I used the reconstruction loss only for training P_y and P_x since we have "ground-truth" from temporal ordering. I added an adversarial loss when using P_y and P_x while going in the spatiotemporal cycle (doing the translation). This was done to provide more constraints because we are effectively using the P_y and P_x on a different distribution than they were trained on (we are using P_x and P_y on generated x and y samples respectively but they were trained on real x and y samples). 

It is true that having additional GAN loss for training P_y and P_x could be useful in training a better predictor. We can make this as a separate model and that would also help in studying the use of additional adversarial loss on performance for prediction modules.
